cognite/client/utils/_identifier.py

```diff
@@ -32,6 +32,22 @@ class InstanceId:
     def dump(self, camel_case: bool = True, include_instance_type: bool = True) -> dict[str, str]:
         return {"space": self.space, "externalId" if camel_case else "external_id": self.external_id}
 
+    @overload
+    @classmethod
+    def load_if(cls, data: None) -> None: ...
+
+    @overload
+    @classmethod
+    def load_if(cls, data: dict[str, str] | tuple[str, str] | Self) -> Self: ...
+
+    @classmethod
+    def load_if(cls, data: dict[str, str] | tuple[str, str] | Self | None) -> Self | None:
```

Comment:
When do we not want this behaviour? Can we just add it to `load`?

Reply:
Hmm, couldn’t this risk not raising on required-but-missing parts of objects? I’ll keep it in mind for v8, I don’t like the thought of extending the possible return types to include None in a minor version (although that may be okay to do?)